src/aiod/configuration/_config.py

```diff
@@ -11,6 +16,20 @@ class Config:
     access_token: str | None = None
     refresh_token: str | None = None
 
+    _observers: dict[str, set[AttributeObserver]] = field(default_factory=lambda: defaultdict(set))
+
+    def subscribe(self, attribute: str, on_change: AttributeObserver):
+        if on_change not in self._observers[attribute]:
+            self._observers[attribute].add(on_change)
+
+    def __setattr__(self, key, value):
+        old_value = getattr(self, key, None)
+        super().__setattr__(key, value)
+
+        if hasattr(self, "_observers"):
```

Comment:
Is this `if` condition necessary? The `Config` class always has the `_observers` attribute, right?

Reply:
This is required for initialization. The attributes (e.g., `api_base_url`) are assigned in the order they are defined. So while initializing it will call its own `__setattr__` before the `_observers` exist yet (since it is defined last, as it has a default value). We do not need observability during the initialization anyway (you cannot be subscribed), so I simply ignore this aspect during initialization.

Reply:
Ok, got it. 👍